tests/virtualization/universal/waitfor_guests.pm

```diff
@@ -40,10 +40,10 @@ sub add_pci_bridge {
         # Skip if a pci-bridge is already present
         return if (script_run("cat $xml | grep 'controller' | grep 'pci-bridge'") == 0);
 
-        my $regex = '<controller type=.pci. .*model=.pci-root..*>';
-        my $addbefore = '<controller type="pci" model="pci-bridge"/>';
-        # the add_before.sh script inserts a given line before the line matching the given regex
-        assert_script_run("virsh dumpxml $guest | /root/add_before.sh '$regex' '$addbefore' > $xml");
+        my $machine = 'i440fx';
+        # add_pci_bridge.sh script inserts 1 pci-bridge for i400fx machine.
+        assert_script_run("virsh dumpxml $guest > $xml");
+        assert_script_run("/root/add_pci_bridge.sh '$machine' '$xml'");
```

Comment:
Why is it important that this functionality is in bash? Wouldn't be native perl subroutine better?

Reply:
Plus, there is already nice `add_pci_bridge()` subroutine :smile: